Source/DataSources/KmlDataSource.js

```diff
@@ -1474,27 +1628,370 @@ define([
         console.log('KML - Unsupported feature: ' + node.localName);
     }
 
+    var RefreshMode = {
+        INTERVAL : 0,
+        EXPIRE : 1,
+        STOP : 2
+    };
+
+    var scratchCartesian2_1 = new Cartesian2();
+    var scratchCartesian3_1 = new Cartesian3();
+    var scratchCartesian3_2 = new Cartesian3();
+    var scratchCartesian3_3 = new Cartesian3();
+    var scratchCartesian3_4 = new Cartesian3();
+    var scratchCartographic = new Cartographic();
+    var horizonPoints = [new Cartesian3(), new Cartesian3(), new Cartesian3(), new Cartesian3()];
+
+    function computeHorizonQuad(camera) {
```

Comment:
So it's really good when you are looking down when zoomed in so no space is showing, it's good on zoomed in horizon views but the bounds are bigger than they actually need to be. If you are zoomed all the way out where we have space on all 4 corners we just return the whole globe. So for stuff on the ground it works well, which is probably the majority of KML use cases. In the general case it falls apart a bit.